### ISSUE TITLE
Focused Launch: Designs for displaying errors

### ISSUE THREAD

**Issue by StefanNieuwenhuis** [Automattic/wp-calypso#48165]
> Part of [#47782](https://github.com/Automattic/wp-calypso/issues/47782)

The user needs to be notified when something goes wrong in the Focused Launch Flow. We cannot allow the user to be stuck on a non-responsive screen, not knowing what to do or what its website's launch status is.

In order to do this, we need to update the UI and for that, we need new designs. 

@ollierozdarz can you help us with designs for error handling?

**Comment by ollierozdarz** [comment]
@StefanNieuwenhuis I can provide a design that handles errors, but are you able to please provide some more information on how/where this might happen in the flow? Ie. is retrieving domains, failure to launch, etc.

**Comment by StefanNieuwenhuis** [comment]
@ollierozdarz 
We basically want to handle every scenario where a user can get "stuck" in a non-responsive screen. This happens mostly when data needs to be retrieved and the connection is interrupted/lost. The main question is how to communicate errors to the user. 

#### Domain Suggestions
One way we're currently handling errors is to display the following banner. This is displayed when something happens that prevents domain suggestions to be retrieved (e.g. no internet). Is this the right way to go? 

<img width="692" alt="Screenshot 2020-12-11 at 13 49 08" src="https://user-images.githubusercontent.com/12104589/101906396-2c1cb700-3bb9-11eb-8300-df15dca67312.png">

#### Unhandled scenario 1: Launch while being offline
When the launch button is clicked and the connection is lost, the user gets stuck on the following screen:
<img width="577" alt="Screenshot 2020-12-11 at 14 01 11" src="https://user-images.githubusercontent.com/12104589/101906546-67b78100-3bb9-11eb-82dd-c6ff913d6936.png">

The user is unaware of the status and needs to be informed of the failing process(es).

#### Unhandled scenario 2: Review plans overview while being offline (Step-by-Step Flow)
Here's another unhandled scenario. When a user enters the step-by-step flow and loses its connection while retrieving the plans overview it gets stuck in a non-responsive screen.

<img width="784" alt="Screenshot 2020-12-11 at 14 18 39" src="https://user-images.githubusercontent.com/12104589/101908197-00e79700-3bbc-11eb-9c7d-183825d82e3d.png">

Same here. The user is stuck and doesn't know what is going on. 

#### Unhandled scenario 3: Checkout after selecting a paid domain + paid plan
Basically the same as scenario 2. When the user clicks on the launch button he's redirected to the checkout page but if the connection is lost the user gets stuck in a non-responsive screen.

<img width="982" alt="Screenshot 2020-12-11 at 14 58 53" src="https://user-images.githubusercontent.com/12104589/101912160-a9e4c080-3bc1-11eb-8576-9e3091f869bc.png">


I hope you get an idea of the different scenarios. @razvanpapadopol do you have any additions? :)

**Comment by razvanpapadopol** [comment]
@StefanNieuwenhuis, **Unhandled scenario 2** should be moved to a separate issue `New Onboarding Launch - error handling` and added to the backlog of https://github.com/orgs/Automattic/projects/127

During the Focused Launch flow, this is how **Plan Details View** looks when site data fails to load:
<img width="1104" alt="Screenshot 2020-12-15 at 15 58 19" src="https://user-images.githubusercontent.com/14192054/102224419-9a7db400-3eee-11eb-93e1-9feab4f33260.png">
 
And this is how **Summary View** looks like:
<img width="1119" alt="Screenshot 2020-12-15 at 16 02 32" src="https://user-images.githubusercontent.com/14192054/102224731-ff390e80-3eee-11eb-9f5c-4599effffd2b.png">

**Comment by StefanNieuwenhuis** [comment]
> @StefanNieuwenhuis, **Unhandled scenario 2** should be moved to a separate issue `New Onboarding Launch - error handling` and added to the backlog of https://github.com/orgs/Automattic/projects/127

Done. See #48401

**Comment by richtabor** [comment]
Closing this as it's quite dated. Thanks.
